ecs-cli/modules/cli/regcreds/regcreds_app.go

```diff
@@ -31,7 +46,184 @@ func Up(c *cli.Context) {
 	if err != nil {
 		log.Fatal("Error executing 'up': ", err)
 	}
-	log.Infof("Read creds input: %+v", credsInput) // remove after SDK calls added
 
-	//TODO: create secrets, create role, produce output
+	err = validateCredsInput(*credsInput)
+	if err != nil {
+		log.Fatal("Error executing 'up': ", err)
+	}
+
+	commandConfig := getNewCommandConfig(c)
+	updateAllowed := c.Bool(flags.UpdateExistingSecretsFlag)
+
+	smClient := secretsClient.NewSecretsManagerClient(commandConfig)
+
+	_, err = getOrCreateRegistryCredentials(credsInput.RegistryCredentials, smClient, updateAllowed)
+	if err != nil {
+		log.Fatal("Error executing 'up': ", err)
+	}
+
+	//TODO: use getOrCreate...() result to create role, produce output
+}
+
+func getOrCreateRegistryCredentials(entryMap readers.RegistryCreds, smClient secretsClient.SMClient, updateAllowed bool) (*map[string]CredsOutputEntry, error) {
+	registryResults := make(map[string]CredsOutputEntry)
+
+	for registryName, credentialEntry := range entryMap {
+		hasCredPair := hasCredPair(credentialEntry)
+		hasSecretARN := false
+		if credentialEntry.SecretManagerARN != "" {
+			hasSecretARN = true
+		}
+
+		log.Infof("Processing credentials for registry %s...", registryName)
+
+		if hasCredPair && hasSecretARN {
+			arn, err := updateOrWarnForExistingSecret(credentialEntry, updateAllowed, smClient)
+			if err != nil {
+				return nil, err
+			}
+			registryResults[registryName] = buildOutputEntry(arn, credentialEntry.ContainerNames)
+
+		} else if hasSecretARN {
+			registryResults[registryName] = buildOutputEntry(credentialEntry.SecretManagerARN, credentialEntry.ContainerNames)
+			log.Infof("Using existing secret %s.", credentialEntry.SecretManagerARN)
+
+		} else {
+			arn, err := createNewRegistrySecret(registryName, credentialEntry, smClient)
+			if err != nil {
+				return nil, err
+			}
+			registryResults[registryName] = buildOutputEntry(arn, credentialEntry.ContainerNames)
+		}
+	}
+
+	return &registryResults, nil
+}
```

Comment:
What do you think about this refactor:
```
func getOrCreateRegistryCredentials(entryMap readers.RegistryCreds, smClient secretsClient.SMClient, updateAllowed bool) (*map[string]CredsOutputEntry, error) {
	registryResults := make(map[string]CredsOutputEntry)

	for registryName, credentialEntry := range entryMap {
		log.Infof("Processing credentials for registry %s...", registryName)

		arn := credentialEntry.SecretManagerARN
		if arn == "" {
			newSecretArn, err := createNewRegistrySecret(registryName, credentialEntry, smClient)
			if err != nil {
				return nil, err
			}
			arn = newSecretArn
		} else if hasCredPair(credentialEntry) {
			if err := updateOrWarnForExistingSecret(credentialEntry, updateAllowed, smClient)
; err != nil {
				return nil, err
			}
		} else {
			log.Infof("Using existing secret %s.", arn)
		}
		registryResults[registryName] = buildOutputEntry(arn, credentialEntry.ContainerNames)
	}

	return &registryResults, nil
}
```

This would also involve changing `updateOrWarnForExistingSecret` to only return an error, e.g.:
```
func updateOrWarnForExistingSecret(credEntry readers.RegistryCredEntry, updateAllowed bool, smClient secretsClient.SMClient) error {
	secretArn := credEntry.SecretManagerARN

	if updateAllowed {
		updatedSecretString := generateSecretString(credEntry.Username, credEntry.Password)
		putSecretValueRequest := secretsmanager.PutSecretValueInput{
			SecretId:     aws.String(secretArn),
			SecretString: updatedSecretString,
		}

		_, err := smClient.PutSecretValue(putSecretValueRequest)
		if err != nil {
			return err
		}

		log.Infof("Updated existing secret %s with new value", secretArn)

	} else {
		log.Warnf("'username' and 'password' found but ignored for existing secret %s. To update existing secrets with new values, use '--update-existing-secrets' flag.", secretArn)
	}

	return nil
}
```